### PR TITLE
fixed sampling random instructions

### PR DESCRIPTION
**Pull request by tzufgoogle**
1. former version of sample generation was with a random seed that made the samples the same.
2. removed seed in case of more than one sample

### CODE REVIEW

File: cabby/geo/walk.py [hunk 1]
```diff
@@ -541,8 +581,10 @@ def generate_and_save_rvs_routes(path: Text, map: map_structure.Map, n_samples:
     columns=['osmid', 'geometry', 'main_tag'])
 
   counter = 0
+  if n_samples > 1:
+    sample_rand = True
```
Comment (jasonbaldridge) [inline cabby/geo/walk.py:585]: This doesn't seem like a clear contract with callers of this function (e.g. I might want a single sample, but to have it picked randomly). I would add an argument that allows them to  specify that they want a deterministic output.

**Review by jasonbaldridge — CHANGES_REQUESTED** [review]
Good to have the deterministic option here! The main thing for this PR is that I think it makes sense to write a Walker class that captures the random vs non-random behavior and applies it consistently to all the sampling.

There are also a lot of function argument lists that are not formatted properly and need to be fixed.

File: cabby/geo/walk.py [hunk 2]
```diff
@@ -90,19 +93,38 @@ def get_end_poi(map: map_structure.Map) -> Optional[GeoSeries]:
...
   # Pick random POI.
-  poi = small_poi.sample(1, random_state=1).iloc[0]
+
+  poi = get_random_sample(small_poi, rand_sample)
   poi['geometry'] = poi.centroid
 
   return poi
 
 
+def get_random_sample(
```
[the remark this answers is not in the document]
Comment (tzufgoogle) [reply cabby/geo/walk.py:103]: Changed

File: cabby/geo/walk.py [hunk 3]
```diff
@@ -75,10 +75,13 @@ def compute_route(start_point: Point, end_point: Point, graph: nx.MultiDiGraph,
   return route_nodes
 
 
-def get_end_poi(map: map_structure.Map) -> Optional[GeoSeries]:
+def get_end_poi(map: map_structure.Map, 
+                rand_sample: bool = False
+                ) -> Optional[GeoSeries]:
```
[the remark this answers is not in the document]
Comment (tzufgoogle) [reply cabby/geo/walk.py:80]: Changed

File: cabby/geo/walk.py [hunk 4]
```diff
@@ -90,19 +93,38 @@ def get_end_poi(map: map_structure.Map) -> Optional[GeoSeries]:
...
+    return df.sample(1).iloc[0]
+  return df.sample(1, random_state=1).iloc[0]
+
+
 def get_start_poi(
-    map: map_structure.Map, end_point: Dict) -> Optional[GeoSeries]:
+                  map: map_structure.Map, 
+                  end_point: Dict, 
+                  rand_sample: bool = False
+                  ) -> Optional[GeoSeries]:
```
[the remark this answers is not in the document]
Comment (tzufgoogle) [reply cabby/geo/walk.py:121]: Changed

File: cabby/geo/walk.py [hunk 5]
```diff
@@ -142,18 +164,22 @@ def get_start_poi(
...
   # Pick random POI.
-  start_point = small_poi.sample(1, random_state=1).iloc[0]
+  start_point = get_random_sample(small_poi, rand_sample)
   start_point['geometry'] = start_point.centroid
   return start_point
 
 
-def get_landmark_if_tag_exists(gdf: GeoDataFrame, tag: Text, main_tag:
-                 Text, alt_main_tag: Text) -> GeoSeries:
+def get_landmark_if_tag_exists(gdf: GeoDataFrame,
```
[the remark this answers is not in the document]
Comment (tzufgoogle) [reply cabby/geo/walk.py:172]: fixed

File: cabby/geo/walk.py [hunk 6]
```diff
@@ -230,12 +259,13 @@ def get_pivot_near_goal(map: map_structure.Map, end_point: GeoSeries) -> \
 
 
 def get_pivot_along_route(
-    route: GeoDataFrame, map: map_structure.Map) -> \
+    route: GeoDataFrame, map: map_structure.Map, rand_sample: bool = False) -> \
     Optional[GeoSeries]:
```
[the remark this answers is not in the document]
Comment (tzufgoogle) [reply cabby/geo/walk.py:263]: fixed

File: cabby/geo/walk.py [hunk 7]
```diff
@@ -263,13 +293,17 @@ def get_pivot_along_route(
   return main_pivot
 
 
-def get_pivot_beyond_goal(map: map_structure.Map, end_point: GeoSeries,
-              route: GeoDataFrame) -> Optional[GeoSeries]:
+def get_pivot_beyond_goal(map: map_structure.Map, 
+                          end_point: GeoSeries,
+                          route: GeoDataFrame, 
+                          rand_sample: bool = False
```
[the remark this answers is not in the document]
Comment (tzufgoogle) [reply cabby/geo/walk.py:299]: yes, added Walker

File: cabby/geo/walk.py [hunk 8]
```diff
@@ -519,8 +558,9 @@ def get_single_sample(map: map_structure.Map) -> Optional[item.RVSPath]:
   return rvs_path_entity
 
 
-def generate_and_save_rvs_routes(path: Text, map: map_structure.Map, n_samples:
-                 int):
+def generate_and_save_rvs_routes(path: Text, 
+                                map: map_structure.Map, 
+                                n_samples:int):
```
[the remark this answers is not in the document]
Comment (tzufgoogle) [reply cabby/geo/walk.py:563]: fixed

File: cabby/geo/walk.py (hunk 1 above)
Comment (tzufgoogle) [reply cabby/geo/walk.py:585]: Changed

**Comment by tzufgoogle** [comment]
fixed format and added walker

**Review by jasonbaldridge — APPROVED** [review]
LGTM!

**Pull request merged**